src/JustEat.StatsD/IStatsDPublisherExtensions.cs

```diff
@@ -59,8 +105,14 @@ public static void Increment(this IStatsDPublisher publisher, long value, double
         /// <param name="publisher">The <see cref="IStatsDPublisher"/> to publish with.</param>
         /// <param name="value">The value to increment the counter(s) by.</param>
         /// <param name="sampleRate">The sample rate for the counter(s).</param>
+        /// <param name="tags">The tags to publish with the counter.</param>
         /// <param name="buckets">The bucket(s) to increment the counter(s) for.</param>
-        public static void Increment(this IStatsDPublisher publisher, long value, double sampleRate, params string[] buckets)
+        public static void Increment(
+            this IStatsDPublisher publisher,
+            long value,
+            double sampleRate,
+            IDictionary<string, string?>? tags = null,
```

Comment:
I'm not sure this parameter will work well being optional if it is followed by the non-optional `params` argument. I think we'd want to do this via overloads.

Reply:
Changed with overloads.